snips-nlu-ontology-ffi-macros/src/ontology.rs

```diff
@@ -118,66 +117,66 @@ impl From<::Slot> for CSlot {
         };
 
         Self {
-            raw_value: CString::new(input.raw_value).unwrap().into_raw(), // String can not contains 0
+            raw_value: CString::new(input.raw_value).unwrap().into_raw(),
             value: CSlotValue::from(input.value),
             range_start: range.start,
             range_end: range.end,
-            entity: CString::new(input.entity).unwrap().into_raw(), // String can not contains 0
-            slot_name: CString::new(input.slot_name).unwrap().into_raw(), // String can not contains 0
+            entity: CString::new(input.entity).unwrap().into_raw(),
+            slot_name: CString::new(input.slot_name).unwrap().into_raw(),
         }
     }
 }
 
 impl Drop for CSlot {
     fn drop(&mut self) {
-        let _ = unsafe { CString::from_raw(self.raw_value as *mut libc::c_char) };
-        let _ = unsafe { CString::from_raw(self.entity as *mut libc::c_char) };
-        let _ = unsafe { CString::from_raw(self.slot_name as *mut libc::c_char) };
+        take_back_c_string!(self.raw_value);
+        take_back_c_string!(self.entity);
+        take_back_c_string!(self.slot_name);
     }
 }
 
 #[repr(C)]
 #[derive(Debug, PartialEq)]
-pub enum CSlotValueType {
-    CUSTOM = 1,
-    NUMBER = 2,
-    ORDINAL = 3,
-    INSTANTTIME = 4,
-    TIMEINTERVAL = 5,
-    AMOUNTOFMONEY = 6,
-    TEMPERATURE = 7,
-    DURATION = 8,
-    PERCENTAGE = 9,
-}
-
-impl CSlotValueType {
-    pub fn from(slot_value: &::SlotValue) -> Self {
+pub enum SNIPS_SLOT_VALUE_TYPE {
+    SNIPS_SLOT_VALUE_TYPE_CUSTOM = 1,
+    SNIPS_SLOT_VALUE_TYPE_NUMBER = 2,
+    SNIPS_SLOT_VALUE_TYPE_ORDINAL = 3,
+    SNIPS_SLOT_VALUE_TYPE_INSTANTTIME = 4,
+    SNIPS_SLOT_VALUE_TYPE_TIMEINTERVAL = 5,
+    SNIPS_SLOT_VALUE_TYPE_AMOUNTOFMONEY = 6,
+    SNIPS_SLOT_VALUE_TYPE_TEMPERATURE = 7,
+    SNIPS_SLOT_VALUE_TYPE_DURATION = 8,
+    SNIPS_SLOT_VALUE_TYPE_PERCENTAGE = 9,
+}
```

Comment:
Why do we repeat the prefix ?

Reply:
No because in C enums are in the global scope

Reply:
Ok